CRM/MembershipExtras/Hook/Post/EntityFinancialTrxn.php

```diff
@@ -41,13 +41,39 @@ public function updatePaymentPlanStatus() {
 
     $newStatus = $this->generatePaymentPlanNewStatus();
     if ($newStatus !== NULL) {
+      if ($newStatus === 'Completed') {
```

Comment:
you can extract the code here to a new private method to make it more clear and to avoid all of this code nesting : 

```
   if ($newStatus !== NULL) {
      civicrm_api3('ContributionRecur', 'create', [
        'id' => $this->recurContribution['id'],
        'contribution_status_id' => $newStatus,
      ]);
    }

    if ($newStatus === 'Completed') {
       $this->updateSubscriptionLinesEndDate();
     }


==========================================

    private function updateSubscriptionLinesEndDate() {
        $subscriptionLines = $this->getSubscriptionLines();
        foreach($subscriptionLines as $line) {
          if (!empty($line['start_date']) && empty($line['end_date'])) {
            civicrm_api3('ContributionRecurLineItem', 'create', [
              'id' => $line['id'],
              'end_date' => $this->recurContribution['end_date'],
            ]);
          }
         }
       }
```